pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,259 @@
+func getLocalProfileConfig(localServerURL string, iteration int, getconfigCtx *getconfigContext) error {
+
+	log.Tracef("getLocalProfileConfig(%s, %d)", localServerURL, iteration)
+
+	localServerMap, err := prepareLocalProfileServerMap(localServerURL, getconfigCtx)
+	if err != nil {
+		return fmt.Errorf("getLocalProfileConfig: prepareLocalProfileServerMap: %s", err)
+	}
+
+	if len(localServerMap) == 0 {
+		return fmt.Errorf(
+			"getLocalProfileConfig: cannot find any configured local subnets for localServerURL: %s",
+			localServerURL)
+	}
+
+	var errList []string
+	for link, srcIP := range localServerMap {
+		resp, contents, err := zedcloud.SendLocal(zedcloudCtx, link, srcIP, 0, nil)
+		if err != nil {
+			errList = append(errList, fmt.Sprintf("SendLocal: %s", err))
+			continue
+		}
+		if resp.StatusCode == http.StatusOK {
+			if err := validateProtoMessage(link, resp); err != nil {
+				// send something to ledmanager ???
+				errList = append(errList, fmt.Sprintf("validateProtoMessage: resp header error: %s", err))
+				continue
+			}
+			getconfigCtx.localProfileReceived = true
+			err := validateAndSetLocalProfile(contents, getconfigCtx)
+			if err != nil {
+				errList = append(errList, fmt.Sprintf("validateAndSetLocalProfile: %s", err))
+				continue
+			} else {
+				return nil
+			}
+		}
+		errList = append(errList, fmt.Sprintf("getLocalProfileConfig: wrong response status code: %d",
+			resp.StatusCode))
+	}
+	return fmt.Errorf("getLocalProfileConfig: all attempts failed: %s", strings.Join(errList, ";"))
+}
+
+func writeReceivedProtoMessageLocalProfile(contents []byte) {
+	writeProtoMessage(savedLocalProfileFile, contents)
+}
+
+//parseProfile process local and global profile configuration
+//must be called before processing of app instances from config
+func parseProfile(ctx *getconfigContext, config *zconfig.EdgeDevConfig) {
+	log.Functionf("parseProfile start: globalProfile: %s currentProfile: %s",
+		ctx.globalProfile, ctx.currentProfile)
+	ctx.globalProfile = config.GlobalProfile
+	if ctx.localProfileServer != config.LocalProfileServer {
+		log.Noticef("parseProfile: LocalProfileServer changed from %s to %s",
+			ctx.localProfileServer, config.LocalProfileServer)
+		ctx.localProfileServer = config.LocalProfileServer
+		if ctx.localProfileReceived {
+			// if we received local profile from another localProfileServer, remove it
+			cleanSavedProtoMessage(savedLocalProfileFile)
+		}
+	}
+	ctx.profileServerToken = config.ProfileServerToken
+	if config.LocalProfileServer == "" {
+		// if we do not use LocalProfileServer set profile to GlobalProfile
+		ctx.currentProfile = config.GlobalProfile
+	} else {
+		if !ctx.readSavedLocalProfile && !ctx.localProfileReceived {
+			// try to read saved local profile actual app may be loaded later
+			if err := readSavedLocalProfile(ctx); err != nil {
+				log.Functionf("parseConfig readSavedLocalProfile: %s", err)
+			}
+		}
```

Comment:
This doesn't belong in the parse function, and because of that I don't understand under what condition we do the readSaved.
Can't the code be structured the same as the readSaved logic for the EdgeDevConfig where the logic is at the top?